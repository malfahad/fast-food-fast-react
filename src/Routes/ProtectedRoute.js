import React from 'react';
import {Redirect,Route} from 'react-router-dom';


const ProtectedRoute = (props)=>{
    if(localStorage.getItem('token'))
    return <Route {...props}/>
    else
    return <Redirect to={props.redirectPath}/>;
}

export default ProtectedRoute;