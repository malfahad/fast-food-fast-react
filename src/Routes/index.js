import React from 'react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import Login from '../components/Login';
import Menu from '../components/Menu';
import Orders from '../components/Orders';
import ProtectedRoute from './ProtectedRoute';
import Register from '../components/Register';

const Routes = () => (
  <Router>
    <Switch>
      <Route path="/login" component={Login} />
      <Route path="/admin/login" component={Login} />
      <Route path="/signup" component={Register} />
      <ProtectedRoute path="/" redirectPath="/login" exact component={Menu} />
      <ProtectedRoute path="/orders" redirectPath="/login" exact component={Orders} />
    </Switch>
  </Router>
);

export default Routes;
