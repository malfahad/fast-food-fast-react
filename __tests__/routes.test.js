import React from 'react';
import { shallow } from 'enzyme';
import Routes from '../src/Routes/index';
import ProtectedRoute from '../src/Routes/ProtectedRoute';

describe('Routes', () => {
  it('should render routes without crashing', () => {
    const wrapper = shallow(<Routes />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render protected routes without crashing', () => {
    const wrapper = shallow(<ProtectedRoute path="/" redirectPath="/login" />);
    expect(wrapper).toMatchSnapshot();
  });
});
