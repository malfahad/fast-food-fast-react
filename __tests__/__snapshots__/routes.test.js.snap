// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routes should render protected routes without crashing 1`] = `ShallowWrapper {}`;

exports[`Routes should render routes without crashing 1`] = `ShallowWrapper {}`;
